pkg/cmd/provisioning/nutanix/create_shared_secrets.go

```diff
@@ -19,36 +23,56 @@ const (
 	manifestsDirName = "manifests"
 
 	secretManifestsTemplate = `apiVersion: v1
+data:
+  credentials: %s
 kind: Secret
 metadata:
-   name: %s
-   namespace: %s
-type: Opaque
-data:
-  NUTANIX_ENDPOINT: %s
-  NUTANIX_PORT: %s
-  NUTANIX_USER: %s
-  NUTANIX_PASSWORD: %s`
-
-	endpointKeyEnvVar = "NUTANIX_ENDPOINT"
-	usernameKeyEnvVar = "NUTANIX_USER"
-	passwordKeyEnvVar = "NUTANIX_PASSWORD"
-	portKeyEnvVar     = "NUTANIX_PORT"
+  name: %s
+  namespace: %s
+type: Opaque`
+
+	BasicAuthCredentialType CredentialType = "basic_auth"
 )
 
-type Credentials struct {
-	NutanixEndpoint string `json:"NUTANIX_ENDPOINT"`
-	NutanixUser     string `json:"NUTANIX_USER"`
-	NutanixPassword string `json:"NUTANIX_PASSWORD"`
-	NutanixPort     string `json:"NUTANIX_PORT"`
+type CredentialType string
+
+type NutanixCredentials struct {
+	Credentials []Credential `json:"credentials"`
+}
+
+type Credential struct {
+	Type CredentialType           `json:"type"`
+	Data *k8sruntime.RawExtension `json:"data"`
+}
+
+type BasicAuthCredential struct {
+	// The Basic Auth (username, password) for the Prism Central
+	PrismCentral PrismCentralBasicAuth `json:"prismCentral"`
+
+	// The Basic Auth (username, password) for the Prism Elements (clusters).
+	// Currently only one Prism Element (cluster) is used for each openshift cluster.
+	// Later this may spread to multiple Prism Element (cluster).
+	PrismElements []PrismElementBasicAuth `json:"prismElements"`
+}
+
+type PrismCentralBasicAuth struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+type PrismElementBasicAuth struct {
+	Name     string `json:"name"`
```

Comment:
How is `name` different from `username` for prism element? Some docs here will be helpful

Reply:
Added the comment above the 'Name' field.